code/game/machinery/turrets/turrets.dm

```diff
@@ -0,0 +1,225 @@
+// Base type for rewritten turrets, designed to hopefully be cleaner and more fun to use and fight against.
+// Use the subtypes for 'real' turrets.
+
+/obj/machinery/turret
+	name = "don't use me"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "turret_barrel"
+	anchored = TRUE
+	idle_power_usage = 500
+	active_power_usage = 60000 // This determines how fast the internal cell gets charged.
+
+	transform_animate_time = 0.2 SECONDS
+
+	// Visuals.
+	var/image/turret_stand = null
+	var/image/turret_ray = null
+	var/ray_color = "#FFFFFFFF" // Color of the ray, changed by the FSM when switching states.
```

Comment:
This will fail unit tests I believe, internally BYOND uses all lowercase for colours so the code quality checks fail on uppercase letters in colour strings.